feature: add tika server support

tomdickman opened the pull request:
Add basic support for the use of a Tika server via REST API calls as per issue #1

Commit:
wip: Add new settings for various tika service types #1

Commit:
wip: add config setting check logic

Commit:
wip: add local_aws dependency for guzzle and future aws support

Commit:
wip: add support for tika server

Commit:
wip: add tests for tika server

Commit:
wip: update README to reflect changes

brendanheywood commented on version.php:
```diff
@@ -24,8 +24,10 @@
-
 $plugin->maturity = MATURITY_ALPHA;
+$plugin->dependencies = array(
+    'local_aws' => 2017030100
```
Everything looks good except this, I'd make it a soft dynamic dependency. Otherwise if you install the metadata plugin but only want to use some other extractor and not tika you are forced to install aws for no reason. Same model as object.

tomdickman replied:
I have removed the hard dependency and added a soft check in the server instantiation, and skipped tests for the server if it's not installed.

Commit:
wip: remove hard dependency on local_aws

move dependency check into server construction as soft check, to prevent blocking use
when tika is installed locally

Commit:
wip: skip tests if server dependency not installed
